lib/neovim-plugin.nix

```diff
@@ -91,15 +119,27 @@ with lib;
       config =
         let
           cfg = config.${namespace}.${name};
-          extraConfigNamespace = if isColorscheme then "extraConfigLuaPre" else "extraConfigLua";
         in
         mkIf cfg.enable (mkMerge [
           {
             extraPlugins = (optional installPackage cfg.package) ++ extraPlugins;
             inherit extraPackages;
           }
-          (optionalAttrs callSetup {
-            ${extraConfigNamespace} = ''
+          (optionalAttrs (!skipConfigAttrs) {
+            ${namespace}.${name}.config.final = ''
+              ${cfg.config.pre}
+              ${cfg.config.init}
+              ${cfg.config.post}
+            '';
+          })
+          (optionalAttrs (callSetup && !skipConfigAttrs) {
+            ${namespace}.${name}.config.init = ''
+              require('${luaName}')${setup}(${optionalString (cfg ? settings) (helpers.toLuaObject cfg.settings)})
+            '';
+            ${configLocation} = cfg.config.final;
+          })
```

Comment:
This should be split up, assigning `config.final` to `configLocation` shouldn't be gated behind `callSetup`:

```nix
(optionalAttrs (callSetup && !skipConfigAttrs) {
  ${namespace}.${name}.config.init = ''
    require('${luaName}')${setup}(${optionalString (cfg ? settings) (helpers.toLuaObject cfg.settings)})
  '';
})
(optionalAttrs (!skipConfigAttrs) {
  ${configLocation} = cfg.config.final;
})
```

Also, we can use `setAttrByPath` if preferred, see above.